sympy/matrices/__init__.py

```diff
@@ -25,11 +26,28 @@
     MatrixSlice, BlockDiagMatrix, BlockMatrix, FunctionMatrix, Identity,
     Inverse, MatAdd, MatMul, MatPow, MatrixExpr, MatrixSymbol, Trace,
     Transpose, ZeroMatrix, OneMatrix, blockcut, block_collapse, matrix_symbols, Adjoint,
-    hadamard_product, HadamardProduct, HadamardPower, Determinant, det,
+    hadamard_product, HadamardProduct, HadamardPower, Determinant,
     diagonalize_vector, DiagMatrix, DiagonalMatrix, DiagonalOf, trace,
     DotProduct, kronecker_product, KroneckerProduct,
     PermutationMatrix, MatrixPermute)
 
+from .determinant import (
+    adjugate, charpoly, cofactor, cofactor_matrix,
+    det, det_bareiss, det_berkowitz, det_LU, minor, minor_submatrix)
```

Comment:
Out of these I am not sure which we want to export.

Currently we have `det` at top-level. We don't want to have two different functions with the same name. Ideally top-level functions like that should work with both `MatrixSymbol` and `Matrix` whenever it makes sense. Of course there should be a function for computing determinants of explicit matrices but it doesn't need to be exported. Functions like `det_bareiss` can just be in the `determinant` module. Users who want those can import them from there. The doc-string for top-level `det` can refer to them or the determinant module to make them more discoverable.

I think `adjugate` and `charpoly` can make sense for `MatrixSymbol` as well (even if not currently implemented) so we would probably want to handle them the same way as `det`. I don't know if the same would apply for `minor` or `minor_submatrix`.

Reply:
`matrices.expressions.determinant.det()` could delegate to `matrices.determinant.det()` if it receives a concrete matrix as an argument.

Reply:
It would be better to do it in `Determinant.doit()` I think.

Reply:
Then you can not pass arguments like `method`, `iszerofunc` and `dotprodsimp`.

Reply:
Most users don't want to pass those arguments. They should be reserved for specialised cases.